azure/scope/identity.go

```diff
@@ -40,7 +37,6 @@ const AzureSecretKey = "clientSecret"
 
 // CredentialsProvider defines the behavior for azure identity based credential providers.
 type CredentialsProvider interface {
-	GetAuthorizer(ctx context.Context, tokenCredential azcore.TokenCredential, tokenAudience string) (autorest.Authorizer, error)
```

Comment:
My dumb grep also shows this which isn't strictly related I don't think, but is also unused AFAICT:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d139d07a2233e5a08dc37ce4a088c4d79e1965ec/util/azure/azure.go#L75-L76

Reply:
also this? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d139d07a2233e5a08dc37ce4a088c4d79e1965ec/util/azure/azure.go#L52